drivers/Serial.cpp

```diff
@@ -36,12 +37,42 @@ int Serial::_putc(int c) {
     return _base_putc(c);
 }
 
+int Serial::getc() {
+    // Mutex is already held
+	if(!core_util_is_isr_active())
```

Comment:
what is this codeblock achieving - getting getc/putc to be accesible from interrupts? Can you provide a bit more details (a reason to change this) - how Stream::getc() helps here with invokation if it's interrupt active?

Reply:
this is the risk of behavior inconsistency I mentioned in the PR description - I'm adding these because in the original code, when it is in ISR context, calling putc()/getc() in Stream class raises this error:
Error - reading from a file in an ISR or critical section** 
So my change here is to call _base_putc/_base_getc in ISR context.